### PR TITLE
feat(payment): PI-1625 Refactor the existing `Af…

### PR DESCRIPTION
**Pull request by bc-sebastianszafraniec**
…firm` components in Checkout JS to use the new checkout payment integration JS API

## What?
Move Affirm to a separate package

## Why?
To use the new checkout payment integration API

## Testing / Proof
Tested manually
Zrzut ekranu 2024-02-26 o 09 15 52

Test coverage
Zrzut ekranu 2024-02-23 o 15 52 39

https://github.com/bigcommerce/checkout-js/assets/130039975/08a8e1f5-338e-4aba-a139-386e95326f5f

@bigcommerce/team-checkout

### CODE REVIEW

**Review by serhii-tkachenko — APPROVED** [review]
LGTM 👍

**Review by PavlenkoM — APPROVED** [review]

**Review by bc-nick — APPROVED** [review]

File: packages/affirm-integration/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": ["../../.eslintrc.json"]
+}
```
Comment (bc-peng) [inline packages/affirm-integration/.eslintrc.json:3]: Can we fix the `No newline at end of file` sign here and some other files?

File: packages/affirm-integration/project.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "test": {
+      "executor": "@nrwl/jest:jest",
+      "outputs": ["coverage/packages/affirm-integration"],
+      "options": {
+        "jestConfig": "packages/affirm-integration/jest.config.js",
+        "passWithNoTests": true
+      }
+    }
+  },
+  "tags": []
```
Comment (bc-peng) [inline packages/affirm-integration/project.json:22]:
```suggestion
  "tags": [
    "scope:integration",
    "scope:shared"
    ]
```

File: packages/affirm-integration/project.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "root": "packages/affirm-integration",
+  "sourceRoot": "packages/affirm-integration/src",
+  "projectType": "library",
+  "targets": {
+    "lint": {
+      "executor": "@nrwl/linter:eslint",
+      "outputs": ["{options.outputFile}"],
+      "options": {
+        "lintFilePatterns": ["packages/affirm-integration/**/*.ts"]
```
Comment (bc-peng) [inline packages/affirm-integration/project.json:10]:
```suggestion
        "lintFilePatterns": ["packages/affirm-integration/**/*.{ts,tsx}"]
```

File: packages/affirm-integration/project.json (hunk 2 above)
Comment (bc-sebastianszafraniec) [reply packages/affirm-integration/project.json]: @bc-peng according to discussion in` #payment-integration-frontend` slack channel https://bigcommerce.slack.com/archives/C02TV6E3BM5/p1709063130646769 I will leave `scope:integration` only ok?

**Review by vitalii-koshovyi — APPROVED** [review]

**Review by bc-peng — APPROVED** [review]

**Pull request merged**